### ISSUE TITLE
How to use virtual appliance

### ISSUE THREAD

**Issue by y0sh188** [mnemonic-no/act#4]
I deployed virtual appliance and accessed to WebUI, but I got this error messages.

`{"responseCode":404,"limit":0,"count":0,"messages":[{"type":"ActionError","message":"Requested URL does not exist.","messageTemplate":"url.not.exist","field":null,"parameter":null,"timestamp":"2019-07-31T08:06:06.697Z"}],"data":null,"size":0}`

And ssh service is not port forwarded to 2222.

I'd like to know how to access WebUI and other services.

**Comment by frbor** [comment]
Hi y0sh188,

have you imported the appliance to VMWare? I did some testing before my vacation and had some problems with vmware, but the documentation is not updated with my findings.

If using vmware you will need to configure networking yourself. You can then decide to either use port forwarding for ssh or connect directly if you have configure a  separate network for your virtual host.

In addition you might need to edit ~act/conf/application.properties and add your IP to the list of allowed hostnames in the setting `cors.allowed.origin`. 

If you are using Virtual Box, I'm not sure why you get this error, so we need to dig into it more closely.

best regards,

   Fredrik

**Comment by y0sh188** [comment]
Hi frbor,

I deployed it to KVM, but it is the same story to need to configure networking.

I accessed the appliance using ssh port forwarding, but I got this same error message.
Also  I added my appliance hostname to ~act/conf/application.properties and restarted the act-app service, but I got the same error message.

I will try to deploy new version appliance.

Thanks,

**Comment by y0sh188** [comment]
Hi Fredrik,

I deployed new virtual appliance image on KVM and virtualbox and was able to access to NiFi and Kibana UI. But I could not search the object on WebUI. I got the ipv4 address "209.200.83.43
" as an indicator on Kibana UI and searched the object on WebUI, I got this error message.

`403: Access denied to requested URL.
You either don't have access to any facts relating to the requested object, or it does not exist in the database`

nginx access log is like this.
`vendor="nginx" product="nginx" site="_" dvc_host="act-ap01" server="192.168.8.61" dest_port="80" dest_ip="192.168.8.61" src="172.16.1.51" src_ip="172.16.1.51" user="-" time_local="06/Aug/2019:08:59:42 +0200" protocol="HTTP/1.1" status="200" bytes_out="402" bytes_in="402" http_referer="http://192.168.8.61/" http_user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.1.2 Safari/605.1.15" nginx_version="1.12.2" http_x_forwarded_for="-" http_x_header="-" uri_query="-" uri_path="/v1/object/ipv4/209.200.83.43" http_method="GET" response_time="0.182" cookie="-" request_time="0.182" 
vendor="nginx" product="nginx" site="_" dvc_host="act-ap01" server="192.168.8.61" dest_port="80" dest_ip="192.168.8.61" src="172.16.1.51" src_ip="172.16.1.51" user="-" time_local="06/Aug/2019:08:59:42 +0200" protocol="HTTP/1.1" status="403" bytes_out="244" bytes_in="244" http_referer="http://192.168.8.61/" http_user_agent="Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.1.2 Safari/605.1.15" nginx_version="1.12.2" http_x_forwarded_for="-" http_x_header="-" uri_query="-" uri_path="/v1/object/ipv4/209.200.83.43/facts" http_method="POST" response_time="0.031" cookie="-" request_time="0.031"`

I also modified "cors.allowed.origins" file to add my server ip.

Do you have any thoughts to fix this issue?

Thanks,

**Comment by frbor** [comment]
Hi,

do you get this error message in the GUI or are you running a curl command. And if running curl, can you provide the full command you are running?

The error means that either the IP address does not exist (on any fact that you have access to).

When you say you find it in kibana, it might also be that you find it in the SCIO index? If so it should (eventually) find it's way to act as well, but it could take some time.

best regards,

  Fredrik

**Comment by y0sh188** [comment]
Hi Fredrik,

I got this error message in the GUI, searched the ipv4 object value "209.200.83.43".
I tried to use API to search for facts with object value "209.200.83.43" and got the results like this.

`$ curl -XPOST --header 'Content-Type: application/json'  --header 'ACT-User-ID: 1'  -d '{"objectValue":["209.200.83.43"] }'  http://192.168.8.61/v1/fact/search`

`{"responseCode":200,"limit":25,"count":2,"messages":null,"data":[{"id":"00042fb6-b2c5-4e4c-824d-88f4455bdccd","type":{"id":"920703ae-cfb7-4f06-b9f9-33b6334e6a29","name":"mentions"},"value":"ipv4","inReferenceTo":null,"organization":{"id":"00000000-0000-0000-0000-000000000001","name":"Test Organization 1"},"source":{"id":"00000000-0000-0000-0000-000000000001","name":"Not implemented yet!"},"accessMode":"Public","timestamp":"2019-08-06T03:12:36.340Z","lastSeenTimestamp":"2019-08-06T03:12:36.340Z","sourceObject":{"id":"9ca2c32c-ee7e-4a29-bd4b-8d9089eaeac3","type":{"id":"8a904ac1-b346-4019-9078-64e0ea4846ad","name":"report"},"value":"ed6665beba968d2663d37bf1f723f78ac79cc814ca89562859b5fa2875b1828a"},"destinationObject":{"id":"c31cf540-2eaf-4660-9167-005e512911cb","type":{"id":"495ebbe1-8cd0-408f-8c1f-e81b53191237","name":"ipv4"},"value":"209.200.83.43"},"bidirectionalBinding":false},{"id":"425d546d-2972-448c-a191-5b9b77fb527b","type":{"id":"e49a405a-1715-4b9c-a86f-04a2c0306006","name":"memberOf"},"value":"ipv4Network","inReferenceTo":null,"organization":{"id":"00000000-0000-0000-0000-000000000001","name":"Test Organization 1"},"source":{"id":"00000000-0000-0000-0000-000000000001","name":"Not implemented yet!"},"accessMode":"Public","timestamp":"2019-08-07T00:33:26.650Z","lastSeenTimestamp":"2019-08-07T00:33:26.650Z","sourceObject":{"id":"c31cf540-2eaf-4660-9167-005e512911cb","type":{"id":"495ebbe1-8cd0-408f-8c1f-e81b53191237","name":"ipv4"},"value":"209.200.83.43"},"destinationObject":{"id":"6312f32d-7d10-44fa-8207-ede4b0218f86","type":{"id":"b90f228f-953d-46a1-b025-085dbe133048","name":"ipv4Network"},"value":"209.200.80.0/20"},"bidirectionalBinding":false}],"size":2}`

I could download the content that includes ipv4 object via SCIO API. 

`$curl 'http://localhost:3000/download?id=ed6665beba968d2663d37bf1f723f78ac79cc814ca89562859b5fa2875b1828a'`

 I'd like to know how to find out the object in the SCIO index.

Thanks,

**Comment by frbor** [comment]
Hi, 

since you can successfully get the result manually using curl I do think that this is an issue with the cors settings.

Can you please tell me the settings of `cors.allowed.origins`?

It should be something like this (includes `http://192.168.8.61`):

```
cors.allowed.origins=http://localhost:8888,http://192.168.8.61
```

And the application service must be restarted afterwards (`service act-app restart`)

SCIO is only a data source in this aspect, and does not contain any facts/objects, but you will find the IOCs extracted from documents there.

It is very nice that you are testing this on KVM, and I might try that myself, but please bear with us that you might get into other issues, since the image is mainly tested on Virtual Box.

regards,
    Fredrik

**Comment by y0sh188** [comment]
Hi Fredrik,

 I could fix this issue adding `http://192.168.8.61` to `cors.allowed.origins`.
Thank you for helping me a lot.

Next, I'll look into the features of act platform.

Thanks,
